### ISSUE TITLE
Mana burst (world entity) in shipyard space

### ISSUE THREAD

**Issue by LordDarthDan** [ValkyrienSkies/Valkyrien-Skies-2#229]
When a mana burst that had previously passed through a Warp lens gets warped with a ship block as target, the resulting burst ends up stuck in shipyard coordinates, and in so doing is no longer able to interact with the "real world".

Mod versions:
VS 2.0.0 alpha4
Botania 1.16.5-420.2

Reproduction:
Build a simple system with a (Botania) mana spreader, force relay, a block and a prism.

Put a Phantom-Warp composite lens on the spreader
Link the relay to the block, observe the burst teleporting to it.
Put an Entropic lens on the prism, observe how the teleported burst is able to explode when touching world blocks.

Now turn the contraption into a ship.
Observe how the burst is no longer visible or able to interact after teleporting (you can only see a faint particle trail).

P.S.: Warp lens uses the entity.setPosition() method: https://github.com/VazkiiMods/Botania/blob/1.16.x-forge/src/main/java/vazkii/botania/common/item/lens/LensWarp.java

**Comment by User-green** [comment]
Do mana spreaders work at all on ships?

**Comment by LordDarthDan** [comment]
> Do mana spreaders work at all on ships?

Yes, they do.
However, since this post in 2022, a change on the Botania side made them harder to operate.

I've tried to find which update changed it, but cannot at the moment.
But basically, up to a point, bursts could not interact with the spreader that fired them.
Now they can - which usually does not happen, but with Valkyrien Skies moving the burst and the spreader, it does.

As a result, there is a very specific method to utilize a mana spreader on a VS ship right now.
You have to break the spreader the tick after it fires the mana burst, which allows the burst to travel normally.

Yes, this wastes the leftover mana. Plan accordingly.

**Comment by User-green** [comment]
> > Do mana spreaders work at all on ships?
> 
> Yes, they do. However, since this post in 2022, a change on the Botania side made them harder to operate.
> 
> I've tried to find which update changed it, but cannot at the moment. But basically, up to a point, bursts could not interact with the spreader that fired them. Now they can - which usually does not happen, but with Valkyrien Skies moving the burst and the spreader, it does.
> 
> As a result, there is a very specific method to utilize a mana spreader on a VS ship right now. You have to break the spreader the tick after it fires the mana burst, which allows the burst to travel normally.
> 
> Yes, this wastes the leftover mana. Plan accordingly.



To be honest, it is just not usable in this state. I hope that this gets resolved, but, well... Years since the issuse has been open and no fix.

**Comment by LordDarthDan** [comment]
> To be honest, it is just not usable in this state. I hope that this gets resolved, but, well... Years since the issuse has been open and no fix.

Well, if slightly involved automation with tradeoffs to work around is your usability floor, so be it.
It's definitely fixable. If only VS wasn't flying through MC versions so fast...

**Comment by blockninja124** [comment]
Valkyrien Skies has been staying on 1.20.1 for the last 2 years, we aren't flying through versions 😭 (much to some peoples disappointment)

**Comment by User-green** [comment]
> Well, if slightly involved automation with tradeoffs to work around is your usability floor, so be it. 

I meant botania on VS, not VS itself! Sorry. (Thoight I *did* drop VS after it got my world corrupted in some kind of a wired shutdown bug I was not able to reproduce)

> t's definitely fixable. If only VS wasn't flying through MC versions so fast...

Understandable.

**Comment by LordDarthDan** [comment]
> botania on VS, not VS itself

I did mean Botania on VS being useable with slightly involve automation - though I suppose my application of Botania on VS is specific (mana cannons, accelerators and radars)...
Perhaps if you're trying to use Botania in a regular everyday way it's a different story.

**Comment by User-green** [comment]
> > botania on VS, not VS itself
> 
> I did mean Botania on VS being useable with slightly involve automation - though I suppose my application of Botania on VS is specific (mana cannons, accelerators and radars)...
> Perhaps if you're trying to use Botania in a regular everyday way it's a different story.

No. I can not transfer mana from generateing flora to pools — thereby bricking the whole mod.

**Comment by LordDarthDan** [comment]
> No. I can not transfer mana from generateing flora to pools — thereby bricking the whole mod.

Yeah... Generating flora on ships is challenging in more than just that one way.
Hence why I had to go with rules of "only storage" and "only sparks".
